server/middleware/productauth.js

```diff
@@ -54,23 +54,32 @@ export default class Productauth{
         return next();
     }
 
-    static async updateproduct(req, res, next) {
-        const { name} = req.body;
-        const { size } = req.body;
-        if (!name) {
+     updateproduct(req, res, next) {
```

Comment:
Did you have to repeat this code? Since it already existed, I thought you would reuse the already existing code